src/tests/e2etest/CommandRunnerTests.cs

```diff
@@ -1,19 +1,279 @@
+                    reportedResult.Wait();
+                    var reportedStatus = reportedResult.Result;
+
+                    if (!condition(reportedStatus))
+                    {
+                        Console.WriteLine("[ExecuteCommandViaCommandRunner] Command status not reported as succeeded for {0}", command.CommandId);
+                        success = false;
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("[ExecuteCommandViaCommandRunner] Exception: {0}", e.Message);
+                success = false;
+            }
+
+            return success;
+        }
+
+        protected async Task<int> SetDesired<T>(string componentName, T value, int ackCode = 200, int maxWaitSeconds = 90)
+        {
+            Twin twin = await _registryManager.GetTwinAsync(_deviceId, _moduleId);
+
+            var twinPatch = new Twin();
+            twinPatch.Properties.Desired[componentName] = value;
+
+            DateTime beforeUpdate = DateTime.Now;
+            Twin updatedTwin = await _registryManager.UpdateTwinAsync(_deviceId, _moduleId, twinPatch, twin.ETag);
+            TwinCollection reported = updatedTwin.Properties.Reported;
 
+            if (!updatedTwin.Properties.Desired.Contains(componentName))
+            {
+                Assert.Warn("{0} not found in desired properties", componentName);
+            }
+
+            DateTime currentTime = DateTime.Now;
+            while (!reported.Contains(componentName) ||
+                   (reported.Contains(componentName) && (reported.GetLastUpdated() < beforeUpdate) && (reported[componentName].GetLastUpdated() < beforeUpdate)))
+            {
+                currentTime = DateTime.Now;
+                if ((currentTime - beforeUpdate).TotalSeconds < maxWaitSeconds)
+                {
+                    await Task.Delay(1000);
+                    updatedTwin = await _registryManager.GetTwinAsync(_deviceId, _moduleId);
+                    reported = updatedTwin.Properties.Reported;
+                }
+                else
+                {
+                    Assert.Warn("Time limit reached while waiting for update to {0} (start: {2} | end: {3} | last updated: {4})", componentName, beforeUpdate, currentTime, reported[componentName].GetLastUpdated());
+                }
+            }
+
+            if (!reported.Contains(componentName))
+            {
+                Assert.Warn("{0} not found in reported properties", componentName);
+            }
+
+            GenericResponse<T> response = JsonSerializer.Deserialize<GenericResponse<T>>(reported[componentName].ToString());
+            return response.ac;
+        }
+
+        protected async Task<int> SetDesired<T>(string componentName, string objectName, T value, int ackCode = 200, int maxWaitSeconds = 90)
```

Comment:
im not crazy about these hardcoded defaults, can we define a global default which is overridable from the pipeline (Environment Var maybe?) and use that as the default?